libs/x-content/src/main/java/org/elasticsearch/common/ParseField.java

```diff
@@ -90,18 +89,18 @@ public ParseField withDeprecation(String... deprecatedNames) {
 
 
     /**
-     * Creates a new field with current name and deprecatedNames, but overrides restApiVersions
-     * @param restApiVersions rest api versions which specifies when a lookup will be allowed
+     * Creates a new field with current name and deprecatedNames, but overrides restApiVersionMatcher
+     * @param restApiVersionMatcher rest api versions which specifies when a lookup will be allowed
      */
-    public ParseField withRestApiVersions(RestApiVersion... restApiVersions) {
-        return new ParseField(this.name, Arrays.asList(restApiVersions), this.deprecatedNames);
+    public ParseField withRestApiVersionMacher(Function<RestApiVersion, Boolean> restApiVersionMatcher) {
```

Comment:
how about `forRestApiVersion( Function<RestApiVersion, Boolean> forRestApiVersion)`

so 

```
PARSER.declareInt(StructWithCompatibleFields::setIntField,
new ParseField("new_name").forRestApiVersion(RestApiVersion.onOrAfter(RestApiVersion.V_8)));
```

Reply:
that looks really good - will rename in all the places accordingly